app/components/primer/popover_component.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Primer
+  class PopoverComponent < Primer::Component
+    CARET_DEFAULT = :none
+    CARET_MAPPINGS = {
+      CARET_DEFAULT => "",
+      :bottom => "Popover-message--bottom",
+      :bottom_right => "Popover-message--bottom-right",
+      :bottom_left => "Popover-message--bottom-left",
+      :left => "Popover-message--left",
+      :left_bottom => "Popover-message--left-bottom",
+      :left_top => "Popover-message--left-top",
+      :right => "Popover-message--right",
+      :right_bottom => "Popover-message--right-bottom",
+      :right_top => "Popover-message--right-top",
+      :top_left => "Popover-message--top-left",
+      :top_right => "Popover-message--top-right"
+    }.freeze
+
+    with_content_areas :heading, :body, :button
```

Comment:
can we change this to `slots`? We are trying to enforce it on new components :)

Reply:
@joelhawksley just suggested using a slot for the message element, so that's switched over now. I'll slotify these other areas, too!